README.md

```diff
@@ -22,9 +39,275 @@ Or install it yourself as:
+
+#### segment_id
+
+`segment_id` allows you to scope permission grants to a specific object id that you define. In our example the `actor` belongs to an Organization model, and we set the organization_id on each permission granted for any actor using the policy.
+
+It provides additional efficiency on query execution, and we can broadly remove permissions if the organization is no longer in the application.
+
+#### unrestricted?
+
+Accessly uses `unrestricted?` to bypass permission checks. This policy shows that the actor has an `admin` designation which we do not want to model in permissions. The business logic implemented here would bypass any permission check if `unrestricted?` returns `true`
+
+### Advanced Action Policy
+
+Let's look at a policy that overrides `action?` and `list` APIs
+
+```ruby
+class UserPolicy < Accessly::Policy::Base
+
+  actions(
+    view: 1,
+    edit_basic_info: 2,
+    change_role: 3,
+    email: 4
+  )
+
+  actions_on_objects(
+    view: 1,
+    edit: 2,
+    destroy: 3
+  )
+
+  def self.namespace
+    User.name
+  end
+
+  def self.model_scope
+    User.all
+  end
+
+  # Override the destroy permission check for an "Action on Object"
+  def destroy?(object)
+    if actor.name == "Alice"
+      true
+    else
+      super
+    end
+  end
+
+  # Override the view permission check for both Action only and "Action on Object"
+  def view?(object = nil)
+    if object.nil?
+      if actor.name == "Bob"
+        false
+      else
+        super
+      end
+    elsif actor.name == "Alice" && object.name == "Bob"
+      true
+    else
+      super
+    end
+  end
+
+  # Override the change_role check for Action only
+  def change_role?
+    false
+  end
+
+  # Override the list method for view permissions
+  def view
+    if actor.name == "Alice"
+      User.all
+    else
+      super
+    end
+  end
+end
+```
+#### Overriding defaults
+
+Here we provide some examples of the `Accessly::Policy::Base` overrides you can make in an application. You can override the function completely or fallback to the `Base` method. The implementation strategy is up to you!
+
+Any call to the following functions will run the given example in the policy:
+
+#### destroy?(object)
+
+```ruby
+# Action on Object queries
+UserPolicy.new(user).destroy?(other_user)
+# or
+UserPolicy.new(user).can?(:destroy, other_user)
+```
+
+#### view?(object = nil)
+
+```ruby
+# Action queries
+UserPolicy.new(user).view?
+# or
+UserPolicy.new(user).can?(:view)
+
+# Action on Object queries
+UserPolicy.new(user).view?(other_user)
+# or
+UserPolicy.new(user).can?(:view, other_user)
+```
+
+#### change_role?
+
+```ruby
+# Action queries
+UserPolicy.new(user).change_role?
+# or
+UserPolicy.new(user).can?(:change_role)
+```
+
+#### view
+
+```ruby
+# List queries
+UserPolicy.new(user).view
+# or
+UserPolicy.new(user).list(:view)
+```
 
 ## Development
```

Comment:
With the idea that we're going to allow PRs from the public, I'm thinking we should remove the `Development` section or at least the second paragraph. Thoughts on that?

Reply:
yea I wasn't sure what to do with these auto-generated sections .. I should have called that out..  I initially leaned towards removing the "##Development" section